packages/vite/src/node/utils.ts

```diff
@@ -863,7 +862,7 @@ export function combineSourcemaps(
     map = remapping(sourcemapList, () => null)
   } else {
     map = remapping(sourcemapList[0], function loader(sourcefile) {
-      if (sourcefile === escapedFilename && sourcemapList[mapIndex]) {
```

Comment:
I'm not really sure what this comparison was here to check. In my repro it's failing with:

```js
{
  sourcefile: './first.ts',
  escapedFilename: '/linux/fixtures/src/pre-bundle/bundle.js'
}

{
  sourcefile: './second.ts',
  escapedFilename: '/linux/fixtures/src/pre-bundle/bundle.js'
}
```

Reply:
That commit was based on source map for Svelte. @benmccann I see you reviewed the PR that merged this line here https://github.com/sveltejs/svelte/pull/5584/files#diff-8742d6c5cbad8aec49f195bc6d25b152ac0cd8180cb876c019f3b74736f9d62aR227. Maybe you have more context on why this was needed?

Reply:
I'm not much of a sourcemap expert. @milahu could know more

From my personal standpoint, I'd just suggest to make sure any changes don't break SvelteKit on the ecosystem CI as we have a small set of sourcemap tests there.